packages/playwright-core/src/server/electron/electron.ts

```diff
@@ -81,20 +86,54 @@ export class ElectronApplication extends SdkObject {
         includeCommandLineAPI: true,
       });
       this._nodeElectronHandlePromise.resolve(new js.JSHandle(this._nodeExecutionContext!, 'object', 'ElectronModule', remoteObject.objectId!));
+      this._nodeChromiumExecutionContext = crExecutionContext;
     });
+    this._nodeSession.on('Runtime.consoleAPICalled', event => this._onConsoleAPI(event));
     this._browserContext.setCustomCloseHandler(async () => {
       await this._browserContext.stopVideoRecording();
       const electronHandle = await this._nodeElectronHandlePromise;
       await electronHandle.evaluate(({ app }) => app.quit()).catch(() => {});
     });
   }
 
+  async _onConsoleAPI(event: Protocol.Runtime.consoleAPICalledPayload) {
+    if (event.executionContextId === 0) {
+      // DevTools protocol stores the last 1000 console messages. These
+      // messages are always reported even for removed execution contexts. In
+      // this case, they are marked with executionContextId = 0 and are
+      // reported upon enabling Runtime agent.
+      //
+      // Ignore these messages since:
+      // - there's no execution context we can use to operate with message
+      //   arguments
+      // - these messages are reported before Playwright clients can subscribe
+      //   to the 'console'
+      //   page event.
+      //
+      // @see https://github.com/GoogleChrome/puppeteer/issues/3865
+      return;
+    }
+    const [crExecutionContext, nodeExecutionContext] = [this._nodeChromiumExecutionContext, this._nodeExecutionContext];
+    if (!crExecutionContext || !nodeExecutionContext) {
+      this._bufferedConsoleMessageEvents.push(event);
+      return;
+    }
+    const args = event.args.map(arg => crExecutionContext.createHandle(nodeExecutionContext, arg));
```

Comment:
I think you don't need `crExecutionContext` at all, just call `nodeExecutionContext.createHandle(arg)`.